src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -786,10 +786,18 @@ private TypeSymbol BindVariableType(CSharpSyntaxNode declarationNode, Diagnostic
             // might own nested scope.
             bool hasErrors = localSymbol.ScopeBinder.ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
 
-            var containingMethod = this.ContainingMemberOrLambda as MethodSymbol;
-            if (containingMethod != null && containingMethod.IsAsync && localSymbol.RefKind != RefKind.None)
+            if (localSymbol.RefKind == RefKind.In)
             {
-                Error(diagnostics, ErrorCode.ERR_BadAsyncLocalType, declarator);
+                var refKeyword = typeSyntax.GetFirstToken();
+                diagnostics.Add(ErrorCode.ERR_UnexpectedToken, refKeyword.GetLocation(), refKeyword.ToString());
```

Comment:
Because binding is NYI and we fallback on "Ref", which is valid. 
I will add PROTOTYPE note

---
In reply to: [102336648](https://github.com/dotnet/roslyn/pull/17223#discussion_r102336648) [](ancestors = 102336648,102323203,102299405)